misc/python/materialize/checks/range.py

```diff
@@ -0,0 +1,110 @@
+                """
+            > DROP MATERIALIZED VIEW IF EXISTS int_range_view1;
+            > DROP MATERIALIZED VIEW IF EXISTS int_range_view2;
+            > DROP TABLE IF EXISTS int_range_table;
+            > CREATE TABLE int_range_table (f1 INT, f2 INT8RANGE);
+            > INSERT INTO int_range_table VALUES (1, '[2,16]'::INT8RANGE);
+            > INSERT INTO int_range_table VALUES (2, NULL);
+            """
+            )
+        )
+
+    def manipulate(self) -> List[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > CREATE MATERIALIZED VIEW int_range_view1 AS SELECT f1, f2 FROM int_range_table;
+                > INSERT INTO int_range_table VALUES (3, '[2,6)'::INT8RANGE);
+                > UPDATE int_range_table SET f2 = '[2,2)'::INT8RANGE WHERE f1 = 3;
+            """,
+                """
+                > CREATE MATERIALIZED VIEW int_range_view2 AS SELECT f1, f2 FROM int_range_table;
+                > INSERT INTO int_range_table VALUES (4, '[2,20]'::INT8RANGE);
+            """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+            > SELECT * FROM int_range_table ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+                        
+            > SELECT * FROM int_range_view1 ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+
+            > SELECT * FROM int_range_view2 ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+
+            > SELECT f1, COALESCE(f2, '[10,80)'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17)
+            2 [10,80)
+            3 empty
+            4 [2,21)
+
+            > SELECT f1, GREATEST(f2, '[2,5]'::INT8RANGE), LEAST(f2, '[2,5]'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) [2,6)
+            2 [2,6) [2,6)
+            3 [2,6) empty
+            4 [2,21) [2,6)
+
+            > SELECT f1, f2, NULLIF(f2, '[2,17)'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) <null>
+            2 <null> <null>
+            3 empty empty
+            4 [2,21) [2,21)
+
+            > SELECT f1, f2, LOWER(f2), UPPER(f2), ISEMPTY(f2), LOWER_INC(f2), UPPER_INC(f2), LOWER_INF(f2), UPPER_INF(f2) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) 2 17 false true false false false
+            2 <null> <null> <null> <null> <null> <null> <null> <null>
+            3 empty <null> <null> true false false false false
+            4 [2,21) 2 21 false true false false false
+
+            > SELECT f1, f2, f2 < '[2,16]'::INT8RANGE AS lt, f2 <= '[2,16]'::INT8RANGE AS le, f2 = '[2,16]'::INT8RANGE AS eq, f2 >= '[2,16]'::INT8RANGE AS ge, f2 > '[2,16]'::INT8RANGE AS gt from int_range_view1 ORDER BY f1 ASC;
```

Comment:
In my opinion, Materialize behaves incorrectly here. At least, the behavior is not in line with Postgres. ⚠️

Materialize:
```
 f1 |   f2   | lt | le | eq | ge | gt 
----+--------+----+----+----+----+----
  1 | [2,17) | t  | t  | t  | t  | t
  2 |        |    |    |    |    | 
  3 | empty  | f  | f  | f  | f  | f
  4 | [2,21) | f  | f  | f  | f  | f
```

Postgres (without printing `f2` column):
```
 f1 | lt | le | eq | ge | gt 
----+----+----+----+----+----
  1 | f  | t  | t  | t  | f
  2 |    |    |    |    | 
  3 | t  | t  | f  | f  | f
  4 | f  | f  | f  | t  | t
```

**Do you agree and shall I create a ticket?**

Reply:
Yes, the inequality seems to be broken. There is no need to involve a table:

```
materialize=> select '[1,1]'::INT8RANGE < '[2,2]'::INT8RANGE;
 ?column? 
----------
 f
(1 row)
```

This returns true in postgres. This is a definite bug, please file it, team Surfaces, assign to Cara and mention Sean in ticket. Tag "T-Correctness" would also be appropriate, as well as mentioning "wrong result" in the title of the ticket.

Reply:
➡️ Created https://github.com/MaterializeInc/materialize/issues/18726